test/boost/secondary_index_test.cc

```diff
@@ -1529,26 +1550,315 @@ SEASTAR_TEST_CASE(test_token_order) {
+
+        eventually(do_tests);
+
+        cquery_nofail(e, "CREATE INDEX t_global ON t(v)");
+
+        eventually(do_tests);
+
+        cquery_nofail(e, "DROP INDEX t_global");
+        cquery_nofail(e, "CREATE INDEX t_local ON t((pk), v)");
+
+        eventually(do_tests);
+    });
+}
+
+SEASTAR_TEST_CASE(test_select_with_token_range_filtering) {
+    return do_with_cql_env_thread([] (auto& e) {
+        cquery_nofail(e, "CREATE TABLE t (pk1 int, pk2 int, ck1 int, ck2 int, v int, v2 int, PRIMARY KEY((pk1, pk2), ck1, ck2))");
+
+         // pk1 | pk2 | system.token(pk1, pk2)
+         // -----+-----+------------------------
+         //   3 |   3 |   -5763496297744201138
+         //   2 |   2 |   -3974863545882308264
+         //   1 |   3 |     793791837967961899
+         //   2 |   1 |    1222388547083740924
+         //   2 |   3 |    3312996362008120679
+         //   1 |   2 |    4881097376275569167
+         //   1 |   1 |    5765203080415074583
+         //   3 |   2 |    6375086356864122089
+         //   3 |   1 |    8740098777817515610
+        for (int pk1 = 1; pk1 <= 3; pk1++) {
+            for (int pk2 = 1; pk2 <= 3; pk2++) {
+                cquery_nofail(e, format("INSERT INTO t(pk1, pk2, ck1, ck2, v, v2) VALUES ({}, {}, 1, 1, 1, 1)", pk1, pk2).c_str());
+                cquery_nofail(e, format("INSERT INTO t(pk1, pk2, ck1, ck2, v, v2) VALUES ({}, {}, 1, 3, 2, 1)", pk1, pk2).c_str());
+                cquery_nofail(e, format("INSERT INTO t(pk1, pk2, ck1, ck2, v, v2) VALUES ({}, {}, 2, 1, 3, 1)", pk1, pk2).c_str());
+            }
+        }
+
+        auto do_tests_ck1 = [&] {
+            auto msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2 FROM t WHERE token(pk1, pk2) >= token(1, 2) AND ck1 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(3) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3) },
+                { int32_type->decompose(3), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(3) }
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 2) AND ck1 = 1 AND v2 = 1 ALLOW FILTERING");
```

Comment:
Again, I'd find this much more readable without repetitive copy-paste.

Reply:
Fixed it in the amended commit.